python-checks/src/test/resources/checks/cdk/unencryptedEbsVolume.py

```diff
@@ -38,3 +38,40 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
 
         volume_args = {"encrypted": True}
         Volume(self, **volume_args)
+
+class SimpleDictValue(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        volume_args = {"encrypted": False}
+        Volume(self, **volume_args) # Noncompliant
+
+class CircularAssignment(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        a = b
+        b = a
+        volume_args = {"encrypted": b}
+        Volume(self, **volume_args)
+
+class DictValueLinkedToVariable(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        a = False
+        volume_args = {"encrypted": a}
+        Volume(self, **volume_args) # Noncompliant
+
+class DictValueLinkedToVariableLinkedToVariable(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        b = False
+        a = b
+        volume_args = {"encrypted": a}
+        Volume(self, **volume_args) # Noncompliant
```

Comment:
Primary and secondary locations in this test case would show, that the trace is correctly represented.